src/Components/MainTableView/DataStorageClasses/TableAttribute.tsx

```diff
@@ -280,35 +280,54 @@ class TableAttribute {
-            <input type="time" step="1" defaultValue={defaultValue} id={tableAttribute.attributeName + "__time"} onChange={(e) => handleChange(e, tableAttribute.attributeName + "__time")}></input>
-          </div>
-        );
+        if (currentValue) {
+          return(
+            <div className="dateTimeFields">
+              <input type="date" defaultValue={currentValue?.split(' ')[0]} id={tableAttribute.attributeName + "__date"} onChange={(e) => handleChange(e, tableAttribute.attributeName + "__date")}></input>
+              <input type="time" step="1" defaultValue={currentValue?.split(' ')[1]} id={tableAttribute.attributeName + "__time"} onChange={(e) => handleChange(e, tableAttribute.attributeName + "__time")}></input>
+            </div>
+          );
+        }
+        else {
+          return(
+            <div className="dateTimeFields">
+              <input type="date" defaultValue={defaultValue} id={tableAttribute.attributeName + "__date"} onChange={(e) => handleChange(e, tableAttribute.attributeName + "__date")}></input>
+              <input type="time" step="1" defaultValue={defaultValue} id={tableAttribute.attributeName + "__time"} onChange={(e) => handleChange(e, tableAttribute.attributeName + "__time")}></input>
+            </div>
+          );
+        }
       }
       else if (tableAttribute.attributeType === TableAttributeType.TIME) {
-        return <input type="text" defaultValue={defaultValue} id={tableAttribute.attributeName} onChange={(e) => handleChange(e, tableAttribute.attributeName)}></input>
-
+        return <input type="text" value={currentValue} defaultValue={defaultValue} id={tableAttribute.attributeName} onChange={(e) => handleChange(e, tableAttribute.attributeName)}></input>
       }
       else if (tableAttribute.attributeType === TableAttributeType.ENUM) {
-        return(
-            <select onChange={(e) => handleChange(e, tableAttribute.attributeName)}> {
+        if (currentValue) {
+          return(
+            <select defaultValue={currentValue} onChange={(e) => handleChange(e, tableAttribute.attributeName)}> {
               tableAttribute.enumOptions?.map((enumOptionString: string) => {
                 return(<option selected={currentValue === enumOptionString} key={enumOptionString} value={enumOptionString}>{enumOptionString}</option>);
               })}
             </select>
         )
+        }
+        else {
+          return(
+            <select defaultValue={defaultValue} onChange={(e) => handleChange(e, tableAttribute.attributeName)}> {
+              tableAttribute.enumOptions?.map((enumOptionString: string) => {
+                return(<option selected={currentValue === enumOptionString} key={enumOptionString} value={enumOptionString}>{enumOptionString}</option>);
+              })}
+            </select>
+        )
+        }
```

Comment:
Uh why is the branch needed here, also I don't think this handle the issuewith no default value correctly